### DISCUSSION TITLE
Does the database connection verify or require SSL?

### DISCUSSION THREAD

**Discussion by paulzakin** [procrastinate-org/procrastinate#755] [Q&A] [answered]
I assume the library follows the directive of PostgreSQL. Like if it has require-ssl, Procrastinate will follow. And it looks like if you also need to verify SSL, you can pass the parameters directly to the underlying database connector, psycopg2. Is that basically right, for both the sync and the async in connector?

Also, quick other question. Does the healthcheck command reopen a new connection on each invocation or reuse the pool?

**Comment by ewjoachim** [comment, marked as answer]
The [Connectors](https://procrastinate.readthedocs.io/en/stable/reference.html#connectors) ref doc spcifies what procrastinate does with the args (so you know exactly what arguments were forwared as-is to what underlying function and what arguments were modified by Procrastinate)

The healthcheck command lives in the procrastinate CLI, which we launch as an autonomous process. The CLI doesn't connect to a local running process of procrastinate. Consequently it executes the same code as other procrastinate processes. There's nothing done for sharing PG connection pools across different processes. (but then, given you have quite some control, on how connectors are setup, e.g. by providing your own pool, you can implement such a mechanism on your side if you want)

**Reply by paulzakin** [reply]
Awesome, thank you!
